### ISSUE TITLE
Infoclick (new version): Problems using text/xml getFeatureInfo-format

### ISSUE THREAD

**Issue by eri-blan** [hajkmap/Hajk#1147]
**Describe the bug**
A few layers we have are served by myCartaServer, which supports the text/xml-format for GetFeatureInfo. In the new infoclick-version, this seems to be broken. myCartaServer does not seem to provide a specific ID-attribute in the response, only the set of attributes and their corresponding values. The error seems to occur in the method below, the id is undefined and therefore the split method throws an error: 

```
  #getLayerNameFromFeatureAndLayer = (feature, layer) => {
    return Object.keys(layer.layersInfo).find((id) => {
      const fid = feature.getId().split(".")[0];
      const layerId = id.split(":").length === 2 ? id.split(":")[1] : id;
      return fid === layerId;
    });
  };
```

I'm not sure if this mainly has something to do with myCartaServers implementation or if it is common for the text/xml getFeatureInfo-format.

A getFeatureInfo response could look like this:

`<?xml version="1.0" encoding="utf-8"?><FeatureCollection xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.mycarta.se/gml" xmlns:gml="http://www.opengis.net/gml" xmlns:xlink="http://www.w3.org/1999/xlink"><gml:featureMember><Hänsyn-Buller><ID>24452</ID><TITEL>Buller skyddsområde</TITEL><BESKRIVNING>Bullerzon för skyttebanor. Slätten är den plats som de kommande 20 åren ska anvisas för skytteverksamhet. Området visar bullerzon inom vilken nybyggnation av bostäder, skolor och liknande inte bör tillåtas. På lång sikt bör skyttebanorna omlokaliseras för att möjliggöra att staden växer. </BESKRIVNING><LINK>http://extra.orebro.se/oversiktsplan/miljorisker.4.38aac5381587bce5d2b15f3.html#op15a4</LINK></Hänsyn-Buller></gml:featureMember></FeatureCollection>`

**To Reproduce**
If it is of interest, we will try to setup a layer on the external test application.

**Comment by jacobwod** [comment]
When I read this, it reminded my of #1090 which was if not the same issue then at least similar. 

Are you sure that you are using a new build? Those this still occur in `develop`?

**Comment by eri-blan** [comment]
> When I read this, it reminded me of #1090 which was if not the same issue then at least similar.
> 
> Are you sure that you are using a new build? Those this still occur in `develop`?

Yes, I also noticed #1090 is similar indeed, but wasn´t sure if the reason for the issue is the same.

I´m using the latest release build (3.10) but I´ll try `develop` and get back to you.

**Comment by jacobwod** [comment]
Can this and #1099 be related?
